src/scripts/main.js

```diff
@@ -1,3 +1,333 @@
+
+            cellArr[cellNum].classList.remove(
+              `field-cell--${celltext}`);
+            cellArr[cellNum + 4].classList.remove(`field-cell--${celltext}`);
+            cellArr[cellNum + 4].innerText = '';
+            scoreNum += Number(celltext) * 2;
+            gameScore.innerText = `${scoreNum}`;
+            accessNum = 1;
+            doublCount++;
+          } else {
+            break;
+          }
+        } else {
+          const celltext = cellArr[cellNum + 4].textContent;
+
+          accessNum = 1;
+          cellArr[cellNum].innerText = celltext;
+
+          cellArr[cellNum].classList.add(
+            `field-cell--${celltext}`);
+          cellArr[cellNum + 4].classList.remove(`field-cell--${celltext}`);
+          cellArr[cellNum + 4].innerText = '';
+        }
+      }
+    });
+
+    cellsChecker();
+
+    if (accessNum === 0) {
+      accessToMoveUp = false;
+    } else {
+      createNewCell();
+      accessToMoveDown = true;
+      accessToMoveRight = true;
+      accessToMoveLeft = true;
+    }
+  }
+
+  if (e.key === 'ArrowDown' && accessToMoveDown) {
+    const rvrsdOccpdCells = [ ...occupiedCells ].reverse();
+    let accessNum = 0;
+
+    rvrsdOccpdCells.forEach(cell => {
+      let cellNum = cellArr.indexOf(cell);
+      let doubleCount = 0;
+
+      while (cellNum <= 16) {
+        cellNum += 4;
+
+        if (cellNum >= 16) {
+          break;
+        } else if (cellArr[cellNum].textContent !== '') {
+          if (cellArr[cellNum].textContent
+            === cellArr[cellNum - 4].textContent
+            && doubleCount === 0) {
+            const celltext = cellArr[cellNum - 4].textContent;
+
+            cellArr[cellNum].classList.add(
+              `field-cell--${Number(celltext) * 2}`);
+            cellArr[cellNum].innerText = `${Number(celltext) * 2}`;
+
+            cellArr[cellNum].classList.remove(
+              `field-cell--${celltext}`);
+            cellArr[cellNum - 4].classList.remove(`field-cell--${celltext}`);
+            cellArr[cellNum - 4].innerText = '';
+            scoreNum += Number(celltext) * 2;
+            gameScore.innerText = `${scoreNum}`;
+            accessNum = 1;
+            doubleCount++;
+          } else {
+            break;
+          }
+        } else {
+          const celltext = cellArr[cellNum - 4].textContent;
+
+          accessNum = 1;
+          cellArr[cellNum].innerText = celltext;
+
+          cellArr[cellNum].classList.add(
+            `field-cell--${celltext}`);
+          cellArr[cellNum - 4].classList.remove(`field-cell--${celltext}`);
+          cellArr[cellNum - 4].innerText = '';
+        }
+      }
+    });
+    cellsChecker();
+
+    if (accessNum === 0) {
+      accessToMoveDown = false;
+    } else {
+      createNewCell();
+      accessToMoveUp = true;
+      accessToMoveRight = true;
+      accessToMoveLeft = true;
+    }
+  }
+
+  if (e.key === 'ArrowRight' && accessToMoveRight) {
```

Comment:
Move all directions to the object.

Example of usage:
`e.key === Directions.ArrowRight`